### DISCUSSION TITLE
Custom Dockerfile for Kubernetes, Why Hasura Server needs database url at startup

### DISCUSSION THREAD

**Discussion by bhanuchhabra7** [hasura/graphql-engine#8775] [General]
I am trying to create a custom dockerfile.yml for the Kubernetes cluster.

The dockerfile.yml looks something like this:

```
FROM hasura/graphql-engine:v2.8.4 as hasura
FROM debian:stable-20220801-slim

#ommited for brevity

ENV HASURA_GRAPHQL_ENABLE_CONSOLE 'true'
ENV HASURA_GRAPHQL_DEV_MODE 'true'

ENTRYPOINT ["graphql-engine","serve"]

#Commented out but tried with both
#CMD ["graphql-engine","serve"]

```

I get an error something like this
--metadata-database-url or --database-url option expected

The question is why Hasura needs a database at startup when I can add sources later on.

**Comment by adas98012** [comment]
Here are the excerpts from. Hasura's Docs:

_Either one of HASURA_GRAPHQL_METADATA_DATABASE_URL or HASURA_GRAPHQL_DATABASE_URL needs to be set with a Postgres database to start a Hasura v2 instance as Hasura always needs a Postgres database to store its metadata._

(Ref: https://hasura.io/docs/latest/guides/upgrade-hasura-v2/#hasura-v2-config-changes)

Hope, this answers your question.
